api/nodes/views.py

```diff
@@ -2358,3 +2359,83 @@ def get_default_queryset(self):
 
     def get_queryset(self):
         return self.get_queryset_from_request()
+
+
+class NodeReorderComponents(JSONAPIBaseView, generics.UpdateAPIView, NodeMixin):
+    """
+    View for Node components reorder.
+
+    PATCH:
+        {
+            "data": [
+                {
+                    "type": "nodes",
+                    "id": <child_node_id>,
+                    "attributes":
+                        {
+                            "_order": <int>
+                        }
+                },
+                {
+                    "type": "nodes",
+                    "id": <child_node_id>,
+                    "attributes":
+                        {
+                            "_order": <int>
+                        }
+                }
+                ]
+        }
+    """
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        IsAdmin,
+    )
+    required_read_scopes = [CoreScopes.NODE_BASE_READ]
+    required_write_scopes = [CoreScopes.NODE_BASE_WRITE]
+
+    view_category = 'nodes'
+    view_name = 'node-reorder-components'
+
+    def get_object(self):
+        return self.get_node()
+
+    def update(self, request, *args, **kwargs):
+        node = self.get_object()
+        node_relations = (
+            node.node_relations
+            .select_related('child')
+            .filter(child__is_deleted=False)
+        )
+        deleted_node_relation_ids = list(
+            node.node_relations.select_related('child')
+            .filter(child__is_deleted=True)
+            .values_list('pk', flat=True),
+        )
+
+        sorted_data = sorted(request.data, key=lambda x: x['_order'])
+
+        new_node_relation_ids = list(node_relations.values_list('id', flat=True))
+        errors = []
+        for node_pos in sorted_data:
+            try:
+                child_node_id = self.get_node(node_id=node_pos.get('id')).id
+                node_relation_obj = node_relations.filter(child_id=child_node_id)
+                if node_relation_obj.exists():
+                    node_relation_id = node_relation_obj.first().id
+                    new_node_relation_ids.remove(node_relation_id)
+                    new_node_relation_ids.insert(node_pos['_order'], node_relation_id)
+            except NotFound:
+                errors.append(node_pos.get('id'))
+
+        if errors:
+            return JsonResponse(
```

Comment:
Probably should use a `ValidationError` from `from rest_framework.exceptions import ValidationError` instead of `JsonResponse`